docs/tools/terraform/reference/cookbook/kafka-flink-integration-recipe.rst

```diff
@@ -1,6 +1,226 @@
-Apache Kafka® as source and sink for Apache Flink® job
-======================================================
+Apache Kafka® as source and sink with Apache Flink® to process streaming data
+=============================================================================
 
-.. Note::
+This example shows how to set up an Aiven for Apache Kafka with an Aiven for Apache Flink integration using the `Aiven Terraform Provider <https://registry.terraform.io/providers/aiven/aiven/latest/docs>`_.
+An Apache Kafka source topic is used as a data source, and Apache Flink processes the data to do filtering or transformation, and finally write the transformed output to a sink topic.
+Please get yourself familiar with :doc:`Aiven for Apache Flink concepts <../../../../products/flink/concepts>` before you start cooking.
 
-    This recipe is under construction due to API changes on Aiven for Apache Flink. Please check back later. 
\ No newline at end of file
+Let's cook!
+-----------
+
+Before looking at the Terraform script, let's visually realize how the services will be connected:
+
+.. mermaid::
+
+   flowchart LR
+      subgraph Aiven for Apache Kafka
+      SourceTopic
+      end
+      subgraph Aiven for Apache Flink Application
+      SourceTopic-->|FlinkTableMapping|FlinkSourceTable
+      FlinkSourceTable-->|Filter/Transform|FlinkTargetTable
```

Comment:
I would put the `Filter/TRansform` as standalone box and have it called `SQL transformation`